### ISSUE TITLE
German translation for error suggestions is sub-optimal

### ISSUE THREAD

**Issue by forki** [Microsoft/visualfsharp#2010]
![temp2](https://cloud.githubusercontent.com/assets/57396/21181854/2da42c8e-c200-11e6-94fb-6cd13f11b25d.png)

It's talking about "Datensatzbezeichnung" which translates to "database records".

Where can we fix that?

**Comment by cartermp** [comment]
@OmarTawfik looks like localization got a little funky in German.  Do you know who we could reach out to?

**Comment by forki** [comment]
I know it's not part of OSS - which is a pitty

**Comment by OmarTawfik** [comment]
@cartermp will send you translation owners info.

**Comment by lukaschal** [comment]
Waiting for linguistic input.

**Comment by forki** [comment]
@lukaschal in this specific scenario we should not translate "record". It's a very well known word in F# context and it's definitely not a database record. 

Is there a chance for me or others to take a look at all the translations? I'm currently trying to improve error messages in the F# compiler and therefore I can also try to help with the german part. I mean the message reads perfect, but it's not 100% technical correct.

**Comment by lukaschal** [comment]
@forki 
Here is a linguistic answer. Is the "correct" suggested translation acceptable ?

Thanks for your message. First, "Datensatzbezeichnung" does by no means refer to "database"—it backtranslates to, well, "record label." If we drop the "record" part, we end up with the following:


SRC: The record label '{0}' is not defined.
TGT: Die Datensatzbezeichnung "{0}" ist nicht definiert.
CORRECT: Die Bezeichnung "{0}" ist nicht definiert.

Problem is, there is hardly any German documentation on F#, so I could not find out, which of the two possible translations ("Bezeichnung" or "Label") is used. Above, I sticked with the (in the MS area) prevalent translation "Bezeichnung" and assume that is fine as it is typically used in programming-language contexts.

**Comment by forki** [comment]
yeah I guess what I'm saying is that record amd record label should probably not translated at all.
They are well established terms in F# and even when I give german talks then I'm talking about "Records". Same with "discriminated unions"

**Comment by lukaschal** [comment]
@forki Linguistic feedback: 

Now I see. The point is not to translate "record" rather than to drop it. So the new (hopefully correct) translation is:  Die Recordbezeichnung "{0}" ist nicht definiert.

Is this translation acceptable ?

**Comment by forki** [comment]
no unfortunately this is also not correct.
The problem now is: "Recordbezeichnung" would basically mean the description of the whole record. But we are talking about a specific field inside the record. these things are called "labels". so the "record label" is something like a "record field".

**Comment by lukaschal** [comment]
@forki Linguist suggested this:

It seems there is no F# terminology yet in German. Therefore, why not call label a "Label" in German if it is actually used? This would lead us to the following translation:

Das Recordlabel "{0}" ist nicht definiert. 

The only problem may be that some people might associate "Recordlabel" with a music-publishing company but the source term has the same problem and that's out of context.

Basically, we might want to establish whether we need an F# terminology on the long run and if yes should act accordingly (that is, set it up).

**Comment by forki** [comment]
"Das Recordlabel "{0}" ist nicht definiert."

I could perfectly live with that. Or "Das Record-Label "{0}" ist nicht definiert."

**Comment by lukaschal** [comment]
@forki  Should be all occurrences of "record label" translated this way ? Should be all strings "label" stay unlocalized on FSharp ? See attached excel sheet for reference.

[fshar-label.xlsx](https://github.com/Microsoft/visualfsharp/files/666120/fshar-label.xlsx)

**Comment by forki** [comment]
Ok I think first step is to make the english error messages consistent.

@dsyme seems we are talking about record fields and record labels. We
should probably just stick to one.

Am 21.12.2016 12:00 schrieb "lukaschal" <notifications@github.com>:

**Comment by cartermp** [comment]
@lukaschal Any updates?

**Comment by lukaschal** [comment]
@forki Is there any action needed from localization for now ? I see last comment "first step is to make the english error messages consistent".

**Comment by forki** [comment]
Yes I think the translation issue is blocked by inconsistent english messages.

**Comment by cartermp** [comment]
Sounds good then.  FWIW I think we should go with Record Labels in the English error messages.

**Comment by dsyme** [comment]
@forki Should we close this, or else submit a PR for the inconsistent english messages?  thx

**Comment by forki** [comment]
I won't too much work

**Comment by dsyme** [comment]
:) Yes, I read the thread about 5 times and thought the same!
